### PR TITLE
[Sprint 4] Feature request off logic #3

### PR DESCRIPTION
**Pull request by tranminhtri911**

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -105,6 +105,10 @@
     <string name="confirm_request_off">Confirm Request Off</string>
     <string name="note">Note: \u0020</string>
     <string name="am_pm">Morning: am - am;  Afternoon: pm - pm;  All day: am - pm</string>
+    <string name="you_can_not_access">You can not access for this month!</string>
```
Comment (daolq3012) [inline app/src/main/res/values/strings.xml]: FATAL: [MissingTranslation] "`you_can_not_access`" is not translated in "ja" (Japanese), "vi" (Vietnamese)
 If an application has more than one locale, then all the strings declared in one language should also be translated in all other languages.

If the string should *not* be translated, you can add the attribute `translatable="false"` on the `<string>` element, or you can define all your non-translatable strings in a resource file called `donottranslate.xml`. Or, you can ignore the issue with a `tools:ignore="MissingTranslation"` attribute.

By default this detector allows regions of a language to just provide a subset of the strings and fall back to the standard language strings. You can require all regions to provide a full translation by setting the environment variable `ANDROID_LINT_COMPLETE_REGIONS`.

You can tell lint (and other tools) which language is the default language in your `res/values/` folder by specifying `tools:locale="languageCode"` for the root `<resources>` element in your resource file. (The `tools` prefix refers to the namespace declaration `http://schemas.android.com/tools`.)

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -105,6 +105,10 @@
...
+    <string name="you_have_to_choose_start_date">You have to choose start date first!</string>
```
Comment (daolq3012) [inline app/src/main/res/values/strings.xml]: FATAL: [MissingTranslation] "`you_have_to_choose_start_date`" is not translated in "ja" (Japanese), "vi" (Vietnamese)
 If an application has more than one locale, then all the strings declared in one language should also be translated in all other languages.

If the string should *not* be translated, you can add the attribute `translatable="false"` on the `<string>` element, or you can define all your non-translatable strings in a resource file called `donottranslate.xml`. Or, you can ignore the issue with a `tools:ignore="MissingTranslation"` attribute.

By default this detector allows regions of a language to just provide a subset of the strings and fall back to the standard language strings. You can require all regions to provide a full translation by setting the environment variable `ANDROID_LINT_COMPLETE_REGIONS`.

You can tell lint (and other tools) which language is the default language in your `res/values/` folder by specifying `tools:locale="languageCode"` for the root `<resources>` element in your resource file. (The `tools` prefix refers to the namespace declaration `http://schemas.android.com/tools`.)

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -105,6 +105,10 @@
...
+    <string name="end_date_must_greater_than_start_day">End day must greater than start day!</string>
```
Comment (daolq3012) [inline app/src/main/res/values/strings.xml]: FATAL: [MissingTranslation] "`end_date_must_greater_than_start_day`" is not translated in "ja" (Japanese), "vi" (Vietnamese)
 If an application has more than one locale, then all the strings declared in one language should also be translated in all other languages.

If the string should *not* be translated, you can add the attribute `translatable="false"` on the `<string>` element, or you can define all your non-translatable strings in a resource file called `donottranslate.xml`. Or, you can ignore the issue with a `tools:ignore="MissingTranslation"` attribute.

By default this detector allows regions of a language to just provide a subset of the strings and fall back to the standard language strings. You can require all regions to provide a full translation by setting the environment variable `ANDROID_LINT_COMPLETE_REGIONS`.

You can tell lint (and other tools) which language is the default language in your `res/values/` folder by specifying `tools:locale="languageCode"` for the root `<resources>` element in your resource file. (The `tools` prefix refers to the namespace declaration `http://schemas.android.com/tools`.)

File: app/src/main/java/com/framgia/wsm/screen/requestoff/RequestOffViewModel.java [hunk 4]
```diff
@@ -161,15 +157,41 @@ public void setPosistionOffType(int posistionOffType) {
...
+    private void validateEndDate(String date) {
+        if (mIsVisibleLayoutNoSalary) {
+            if (DateTimeUtils.convertStringToDate(date)
+                    .before(DateTimeUtils.convertStringToDate(mStartDateNoSalary))) {
+                setEndDate(null);
+                showErrorDialog(mContext.getString(R.string.end_date_must_greater_than_start_day));
+            } else {
+                setEndDate(date);
+            }
+        }
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/requestoff/RequestOffViewModel.java]: ERROR: [PMD-CPD] 44 lines duplicated.
app/src/main/java/com/framgia/wsm/screen/requestleave/RequestLeaveViewModel.java

Comment (tranminhtri911) [inline app/src/main/java/com/framgia/wsm/screen/requestoff/RequestOffViewModel.java]: cái này em chưa viết xong, pull sau mới làm tiếp được

**Pull request merged**